### DISCUSSION TITLE
How do I prevent the "pressed" class from being applied on a listboxitem when clicking a control inside the listboxitem?

### DISCUSSION THREAD

**Discussion by H0tCh0colat3** [AvaloniaUI/Avalonia#17254] [Q&A]
I have a structure like this:
```
ListBox
    ListBoxItem
       Some Clickable Control (that listens for pointer pressed)
```

The pointerpressed event on Some Clickable Control always triggers the pressed class on the parent ListBoxItem. While this doesn't actually select the item, visually, it looks not great due to the color flashing. Ideally, I'd be able to catch this event and handle it if the source is a control within the listboxitem. After further investigation, it seems the code here listens for the event during the tunneling phase.
https://github.com/AvaloniaUI/Avalonia/blob/7f2cd0723a1f5418fcda7e5a922948685c36e2e2/src/Avalonia.Controls/Mixins/PressedMixin.cs#L19
From what I understand (please correct me if I've got it wrong), this prevents any opportunity to stop the event during the bubbling phase, because by the time it reaches the bubbling phase, the class is already applied. I don't think I can handle it during the tunneling phase either, because then the event doesn't propagate to the control I actually want to receive the event. Is there a way to work around this?

**Comment by maxkatz6** [comment]
You are right, tunnel handling makes it harder to "cancel" pressed visuals on the `ListBoxItem`.
But on the other hand, do you need it?
If you need a UI list of clickable controls that you define, you can use `ItemsControl` instead of `ListBox`.
Additionally, you can use `SelectingItemsControl` if you want to keep selection behavior, but you would need to set IsSelected on your own (from the clickable control handler, for example).
